library/core/src/main/java/com/google/android/exoplayer2/extractor/ts/H264Reader.java

```diff
@@ -209,6 +212,43 @@ private void endNalUnit(long position, int offset, int discardPadding, long pesT
   }
 
   /**
+   * ITU-T H.264: 7.3.2.3.1 Supplemental enhancement information message syntax .
+   */
+  private void checkRecoveryPoint(byte[] data, int length) {
+    ParsableNalUnitBitArray bitArray = new ParsableNalUnitBitArray(data, 0, length);
+    int type = 0;
+    int size = 0;
+    int ffByte = 0;
+
+    bitArray.skipBits(24);
+    do {
+      if (!bitArray.canReadBits(8))
+        return;
+      ffByte = bitArray.readBits(8);
+      type += ffByte;
+    } while (ffByte == 255 /* equal to 0xFF */);
+
+    if (type != SEI_TYPE_RECOVERY_POINT)
+      return;
+
+    do {
+      if (!bitArray.canReadBits(8))
+        return;
+      ffByte = bitArray.readBits(8);
+      size += ffByte;
+    } while (ffByte == 255 /* equal to 0xFF */);
+
+    // Read the slice header using the syntax defined in ITU-T Recommendation H.264 (2013)
+    // subsection 7.3.3.
+    if (!bitArray.canReadExpGolombCodedNum()) {
+      return;
+    }
+    int recoveryFrameCount = bitArray.readUnsignedExpGolombCodedInt();
```

Comment:
Are you sure the implementation here is correct? I'm seeing large values being read here (e.g. 223). It doesn't feel right that the number would be so large?